pkg/sql/schema_changer.go

```diff
@@ -137,6 +138,18 @@ func NewSchemaChangerForTesting(
 	}
 }
 
+// IsConstraintError returns true if the error is considered as
+// an error introduced by the user. For example a constraint
+// violation.
+func IsConstraintError(err error) bool {
+	pgCode := pgerror.GetPGCode(err)
+	return pgCode == pgcode.CheckViolation ||
+		pgCode == pgcode.UniqueViolation ||
+		pgCode == pgcode.ForeignKeyViolation ||
+		pgCode == pgcode.NotNullViolation ||
+		pgCode == pgcode.IntegrityConstraintViolation
+}
```

Comment:
Assuming this is correct (I honestly don't know) this might not age so well if we need to cover new failure codes. Have you considered making the pgcode itself part of the telemetry counter key? That way you wouldn't have to worry about this. I believe it'd be OK to shift the burden of interpreting these to the client. cc @vy-ton

Reply:
> pgcode itself part of the telemetry counter key
Do you mean a counter for every time a pgcode is returned? Could that be noisy and include instances when these specific codes are encountered in non-schema changes?

When we have the new telemetry logging infra, I think tracking pgcode errors would be a great thing to add.